ip-geo.html

```diff
@@ -0,0 +1,74 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Health Bot Geo Lookup</title>
+</head>
+
+<body>
+    <h1>Health bot</h1>
+    <p>Hello <span id="user_ip"></span> from <span id="user_countrycode"></span> <span id="user_region"></span></p>
+    <p>The health bot is <span id="user_healthbotenabled"></span> for your country and state.</p>
+    <div id="disabled_message"></div>
+
+    <script type="application/javascript">
+    
+        /* this function is automatically invoked via the geojs script and json object is populated with location information */
+        function geoip(json) {
+            determineBotEnabled(json);
+        }
+
+        /* this function uses ip location information to determine if the health bot should be enabled */        
+        function determineBotEnabled(json) {
+
+            // update page elements with information collected from geojs ip location lookup
+            var userip = document.getElementById("user_ip");
+            var countrycode = document.getElementById("user_countrycode");
+            var region = document.getElementById("user_region");
+            var healthBotEnabled = document.getElementById("user_healthbotenabled");
+            userip.textContent = json.ip;
+            countrycode.textContent = json.country_code;
+            region.textContent = json.region;
+
+            // message to display if the client country and state is disabled
+            var disabledMessageText = "The health bot is not currently available in your area. THIS MESSAGE ONLY SHOWS IF DISABLED BY COUNTRY AND STATE!!!";
+
+            // bool to track whether or not the bot is enabled
+            var enableBot = false;
+
+            // define the list of enabled regions to use the healthbot currently
+            /* MODIFY THIS LIST TO ENABLE HEALTHBOT IN SPECIFIED STATES */
+            var enabledCountry = "US";
+            var enabledStates = ['California', 'Washington', 'New York'];
+
+            try {
+                // lowercase all enbledStates
+                var lowerCaseEnabledStates = enabledStates.map(function (value) {
+                    return value.toLowerCase()
+                });
+
+                // determine if the current client is in one of the enabled states and country
+                var statedEnabled = lowerCaseEnabledStates.indexOf(json.region.toLowerCase());
+                var countryEnabled = enabledCountry.toLowerCase() === json.country_code.toLowerCase();
+
+                // determine if we should show the bot
+                enableBot = statedEnabled === 1 && countryEnabled;
+            } catch (error) {
+                console.log(error);
+            }
+
+            // use the value of enableBot to determine whether or not to display the healthbot
+            if (enableBot) {
```

Comment:
I would handle this outside this function, because I think this function should just return true/false. (aka the healthBotEnabled state)